x/logic/interpreter/fs/parser.go

```diff
@@ -0,0 +1,31 @@
+package fs
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+type URIHandler interface {
+	CanOpen(ctx context.Context, uri *url.URL) bool
+	Open(ctx context.Context, uri *url.URL) ([]byte, error)
```

Comment:
As the file opened may contains information related to its source It could be interesting to return a `os.File` implementation instead, allowing full control of its content on the `URIHandler` side.